nss/aad_auth/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/ubuntu/aad-auth/internal/logger"
+	"github.com/ubuntu/aad-auth/internal/nss"
+)
+
+func main() {
+	flag.Usage = aadAuthUsage
+	flag.Parse()
+
+	switch flag.Arg(0) {
+	case "getent":
+		ctx := nss.CtxWithSyslogLogger(context.Background())
+		defer logger.CloseLoggerFromContext(ctx)
+
+		db, key := flag.Arg(1), flag.Arg(2)
+
+		if !dbIsSupported(db) {
+			log.Fatalf("Request db %s is not supported", db)
+		}
+
+		out := Getent(ctx, db, key)
+		fmt.Print(out)
+	case "":
+		flag.Usage()
+		os.Exit(1)
+	default:
+		log.Fatalf("Invalid argument %s", flag.Arg(0))
+	}
+}
+
+func dbIsSupported(db string) bool {
+	supportedDbs := []string{"group", "passwd", "shadow"}
+	for _, d := range supportedDbs {
+		if d == db {
+			return true
+		}
+	}
+	return false
+}
+
+func aadAuthUsage() {
+	fmt.Fprintln(os.Stderr, `
+This executable should not be used directly, but should you wish too:
+
+Usage: aad_auth getenv {dbName} {key}
+		
+dbName: Name of the database to be queried. Supported values: passwd, group, shadow
```

Comment:
I'm not sure I understand what you mean here.

Reply:
Basically, do not hardcode the dbs and key in the usage string. For ease of maintainance, have all that in a map (see comment below) and reuse that same map in the code to filter supported dbs.

Reply:
Ack. I'm not about about treating the keys though... Since they can be strings (for querying by name), will be a pain to treat them. It's probably best to let it go on and return the output with ErrNoEnt.